### DISCUSSION TITLE
item submission mode

### DISCUSSION THREAD

**Discussion by TomWoodward** [amp-up-io/qti3-item-player-controller#10] [General]
I noticed that the player's data model associates submissionMode with individual items, but it seems to me that this isn't how the data is modeled in the QTI3 spec. it seems like the submission mode in the xml is intended to go in the assessment structure, and that would apply to a group of items? 

but also it seems like its easily possible to have items that don't work if they expect to be using individual submission and you put them in a test that uses simultaneous submission, so i wonder why items don't declare their anticipated/supported submission modes

**Comment by paulgrudnitski** [comment]
In the QTI 3 Item Player Controller sample collections, the submission mode (as well as item session control properties such as validateResponses and showFeedback) for each item is specified in the TestFactory.

```
     {
        "identifier": "card-08a-baseline",
        "sessionControl": {
          "showFeedback": true,
          "validateResponses": false,
          **"submissionMode": "simultaneous"**
        }
      },
      {
        "identifier": "adaptivecard-08c-showexample",
        "sessionControl": {
          "showFeedback": true,
          "validateResponses": false,
          "submissionMode": "simultaneous"
        }
      },
      {
        "identifier": "adaptivecard-08b-solvethis",
        "sessionControl": {
          "showFeedback": true,
          "validateResponses": false,
          "submissionMode": "simultaneous"
        }
      },
```

This is because the QTI 3 Item Player Controller does not understand `<qti-assessment-test>` xml.  This app is intended to be a demonstration of various capabilities of the QTI 3 Item Player component.  

The fully expressive QTI 3 Test Player (Test Explorer), which understands everything in QTI 3, will be available and free and MIT licensed in the coming months.
